core/iwasm/compilation/aot_llvm.c

```diff
@@ -3068,6 +3069,17 @@ aot_create_comp_context(const AOTCompData *comp_data, aot_comp_option_t option)
     }
     LLVMDisposeMessage(triple);
 
+#if WASM_ENABLE_WAMR_COMPILER != 0
+    /* Only if SIMD is actually used, we need to set the compiler
+     * options */
+    if (((WASMModule *)comp_data->wasm_module)->is_simd_used) {
+        option->enable_simd = true;
+    }
+    else {
+        option->enable_simd = false;
```

Comment:
Can just `option->enable_simd = comp_data->wasm_module->is_simd_used;`

A concern is that then `wamrc --disable-simd` is omitted, should we do more for it?

Reply:
Yes in this implementation --disable-simd is omitted, so should we raise error if --disable-simd is set but SIMD is used in the module?

Reply:
Agree, it should be better.